src/Provisioner.js

```
@@ -0,0 +1,341 @@
+    hashtag = hashtag.replace("#", "");
+    return roomstore.getEntriesByRemoteId("hashtag_"+hashtag).then(rooms => {
+      const isLinked = rooms.length > 0;
+
+      if(isLinked) {
+        return {body: "Hashtag already bridged!"};
+      }
+
+      var remote = new RemoteRoom("hashtag_" + hashtag);
+      remote.set("twitter_type", "hashtag");
+      remote.set("twitter_hashtag", hashtag);
+      roomstore.linkRooms(new MatrixRoom(room_id), remote);
+
+      this._twitter.timeline.add_hashtag(hashtag, room_id, true);
+      return {};
+    })
+
+
+  }
+
+  isProvisionRequest (req) {
+    return req.url.match(/^\/_matrix\/provision\/(\S+)\/(link|links|timeline)/)
+      || req.url.startsWith("/_matrix/provision/show");
+  }
+
+  _updateBridgingState (roomId, userId, status, skey) {
+    const intent = this._bridge.getIntent();
+    try {
+      intent.client.sendStateEvent(roomId, 'm.room.bridging', {
+        user_id: userId,
+        status: status // pending, success, failure
+      }, skey);
+    }
+    catch (err) {
+      log.error("Provisioning", "Couldn't update bridging state for %s", roomId );
+      throw new Error(`Could not update m.room.bridging state in this room`);
+    }
+  }
+
+  * _userHasProvisioningPower (self, userId, roomId) {
+    log.info("Provisioning", `Check power level of ${userId} in room ${roomId}`);
+    const matrixClient = self._bridge.getClientFactory().getClientAs();
+
+    // Try to join a room, or timeout after 1 min
+    try {
+      yield matrixClient.joinRoom(roomId).timeout(ROOM_JOIN_TIMEOUT_MS);
+    } catch (e) {
+      return Promise.reject({err: 403, body: "Couldn't join room. Perhaps room permissions are not set to public?"});
+    }
+    var powerState;
+    try{
+      powerState = yield matrixClient.getStateEvent(roomId, 'm.room.power_levels');
+    }
+    catch(err) {
+      log.error("Provisioning", `Error retrieving power levels (${err.data.error})`);
+    }
+
```

This error is dropped to the floor. It should at least be logged.